lib/src/robot_interface/robot_interface.cpp

```diff
@@ -632,15 +671,38 @@ void RobotInterface::setJointCommands(double s0, double s1, double e0, double e1
 
 bool RobotInterface::detectForceInteraction()
 {
-    double f_x = abs(_curr_wrench.force.x - _filt_force[0]);
-    double f_y = abs(_curr_wrench.force.y - _filt_force[1]);
-    double f_z = abs(_curr_wrench.force.z - _filt_force[2]);
+    // ROS_INFO("Filt Forces: %g, %g, %g", _filt_force[0], _filt_force[1], _filt_force[2]);
+    // compare the current force to the filter force. if the difference is above a
+    // threshold defined in utils.h, return true
+
+    double small_thres = exp (-5);
```

Comment:
Also, you keep having these "small values" throughout the code. It would be better to define an EPSILON in `utils.h`, and then use it everywhere you need it.